x-element.js

```diff
@@ -1012,25 +1012,70 @@ export default class XElement extends HTMLElement {
 
 /** Internal implementation details for template engine. */
 class TemplateEngine {
-  static #UNSET = Symbol(); // Ensures a unique, initial comparison.
-  static #ATTRIBUTE_PREFIXES = {
-    attribute: 'x-element-attribute-',
-    boolean: 'x-element-boolean-',
-    property: 'x-element-property-',
-  };
-  static #CONTENT_PREFIX = 'x-element-content-';
-  static #CONTENT_REGEX = new RegExp(`${TemplateEngine.#CONTENT_PREFIX}(\\d+)`);
-  static #OPEN = /<[a-z][a-z0-9-]*(?=\s)/g;
-  static #STEP = /\s+[a-z][a-z0-9-]*(?=[\s>])|\s+[a-z][a-zA-Z0-9-]*="[^"]*"/y;
-  static #ATTRIBUTE = /\s+(\??([a-z][a-zA-Z0-9-]*))="$/y;
-  static #PROPERTY = /\s+\.([a-z][a-zA-Z0-9_]*)="$/y;
-  static #CLOSE = />/g;
-
-  static #interface = null;
-  static #stateMap = new WeakMap(); // Maps nodes to internal state.
-  static #analysisMap = new WeakMap(); // Maps strings to cached computations.
-  static #resultMap = new WeakMap(); // Maps symbols to results.
-  static #updaterMap = new WeakMap(); // Maps symbols to updaters.
+  // Special markers added to markup enabling discovery post-instantiation.
+  static #ATTRIBUTE_PREFIX = 'x-element-attribute';
+  static #BOOLEAN_PREFIX = 'x-element-boolean';
+  static #DEFINED_PREFIX = 'x-element-defined';
+  static #PROPERTY_PREFIX = 'x-element-property';
+  static #CONTENT_PREFIX = 'x-element-content';
+
+  // Patterns to find special edges in original html strings.
+  static #OPEN_REGEX = /<[a-z][a-z0-9-]*(?=\s)/g;
+  static #STEP_REGEX = /(?:\s+[a-z][a-z0-9-]*(?=[\s>])|\s+[a-z][a-zA-Z0-9-]*="[^"]*")+/y;
```

Comment:
Small perf win here. Rather than match multiple times, we can let the single regex match more greedily by wrapping the whole thing in a non-capturing group and adding a `+` to the end.